### DISCUSSION TITLE
Simulate server-side channel failure

### DISCUSSION THREAD

**Discussion by lali1-betsson** [rabbitmq/rabbitmq-server#7644] [Questions]
Sometimes we experience unexpected channel closures, initiated by server.
Sometimes our client-side custom recovery logic fails to properly create another one.

So I'd like to (ideally programmatically) force a server-side channel close, so that I could test our custom recovery logic.
(Closing the whole connection manually in management console is a different scenario, I'd like just to kill a single channel in a working connection).
Any ideas how to do this?

**Comment by lukebakken** [comment]
> Sometimes we experience unexpected channel closures, initiated by server

Please **attach** a log file showing this. "Unexpected" channel closures shouldn't happen with RabbitMQ.

Closing a channel is not an operation we're likely to support. You can use the [HTTP API](https://rawcdn.githack.com/rabbitmq/rabbitmq-server/v3.11.10/deps/rabbitmq_management/priv/www/api/index.html) to close connections.

**Reply by michaelklishin** [reply]
Well, a client still can [trigger a channel exception](https://rabbitmq.com/channels.html#error-handling) by first declaring, say, an exchange as transient and then as durable (under the same name).

As far as RabbitMQ is concerned, that will terminate the channel due to a protocol exception. The
client will get a notification, not sure if this is the scenario @lali1-betsson is looking for.

**Reply by lali1-betsson** [reply]
This happens relatively rarely, so I don't have the server logs unfortunately, we dont keep them for long.
(But next time I will grab it :) )
What I know is that our .net client lib reports channel shutdown event with errorcode 541, for what this is the comment in the client lib code:
"Connection: The server could not complete the method because of an internal error. The server may require intervention by an operator in order to resume normal operations."

Anyway, my question is not about investigating why this error happens.
I'm looking for ways to reproduce deterministically the client lib to shutdown the channel with 541, so that we could create a proven working mitigation for this event.
